backends/tfhe-cuda-backend/cuda/src/device.cu

```diff
@@ -52,184 +50,132 @@ void *cuda_malloc_async(uint64_t size, cuda_stream_t *stream) {
   return ptr;
 }
 
-/// Checks that allocation is valid
-/// 0: valid
-/// -1: invalid, not enough memory in device
-/// -2: invalid, gpu index doesn't exist
-int cuda_check_valid_malloc(uint64_t size, uint32_t gpu_index) {
-
-  if (gpu_index >= cuda_get_number_of_gpus()) {
-    // error code: invalid gpu_index
-    return -2;
-  }
-  cudaSetDevice(gpu_index);
+/// Check that allocation is valid
+void cuda_check_valid_malloc(uint64_t size, uint32_t gpu_index) {
+  check_cuda_error(cudaSetDevice(gpu_index));
   size_t total_mem, free_mem;
-  cudaMemGetInfo(&free_mem, &total_mem);
+  check_cuda_error(cudaMemGetInfo(&free_mem, &total_mem));
   if (size > free_mem) {
-    // error code: not enough memory
-    return -1;
+    printf("Cuda error: not enough memory on device. "
+            "Available: %d vs Requested: %d", free_mem, total_mem);
+    abort();
   }
-  return 0;
 }
 
 /// Returns
 ///  -> 0 if Cooperative Groups is not supported.
 ///  -> 1 otherwise
 int cuda_check_support_cooperative_groups() {
```

Comment:
any reason to return an int rather than a bool here ? 🤔

Reply:
No, I'll change it